js/components/data-map.js

```diff
@@ -41,7 +48,159 @@
+            for (var i = 1; i < steps + 1; i++) {
+              var val = interval
+                ? i / steps
+                : i;
+              arr.push(val);
+            }
+            return arr;
+          }
+
+          function createDomains (interval, domain) {
+            var domainStart = [],
+              domainEnd = [];
+            interval.forEach(function (step, i) {
+              if (i === 0) {
+                domainStart.push(domain[0]);
+              } else {
+                domainStart.push(interval[i - 1] * domain[1]);
+              }
+              domainEnd.push(step * domain[1]);
+            });
+            return domains = [domainStart, domainEnd];
+          }
+
+
+          function uniq(value, index, self) {
+            return self.indexOf(value) === index;
+          }
+
+          function getUnique(data, _steps, domain) {
+            var getSteps = d3.scale[type]()
+              .domain(domain)
+              .range(_steps);
+
+            var values = [];
+            data.forEach(function(d){
+              values.push(getSteps(d));
+            });
+
+            return values.filter(uniq);
+          }
+
+          function filterSteps(uniqueSteps, context) {
+
+            context.querySelectorAll('[legend-swatch]')
+              .forEach(function(s){
+                s.setAttribute('aria-hidden', true);
+              });
+
+            uniqueSteps.forEach(function(step) {
+              var allThresholds = context.querySelectorAll('[legend-swatch]');
+
+              allThresholds.forEach(function(s){
+
+                var diff = s.hasAttribute('legend-threshold')
+                  ? +s.getAttribute('legend-threshold') - +step
+                  : +s.getAttribute('data-step') - +step;
+
+                if (diff === 0) {
+                  s.setAttribute('aria-hidden', false);
+                }
+              });
+            });
+          }
+
+          var intervals = createSteps(steps, true);
+          var _steps = createSteps(steps);
+
+          var uniqueSteps = getUnique(marks.data(), _steps, domain);
+          var domains = createDomains(intervals, domain);
+
+          var textScaleStart = d3.scale[type]()
+            .domain(stepDomain)
+            .range(domains[0]);
+
+          var textScaleEnd = d3.scale[type]()
+            .domain(stepDomain)
+            .range(domains[1]);
+
+          var swatches = d3.select(this)
+            .selectAll('dl [data-step]')
+            .datum(function() {
+              return +this.getAttribute('data-step') || 1;
+            });
+
+          swatches.style('background-color', scaleSteps);
+
+          format.start = function(value, units) {
+            if (units === '$' || units.includes('dollar')) {
+              return [format.dollars(value), ' –'].join(' ');
+            } else {
+              return [format.si(value), ' –'].join(' ');
+            }
+          };
+
+          format.end = function(value, units) {
```

Comment:
I like the idea of moving the units out of the legend. After I did it, I also noticed that long units look bad, and that shorter ones look repetitive, but I thought I would wait for @ericronne's feedback/redesign. 

I'll try to pair things down where I can for now